great_tables/_utils_render_html.py

```diff
@@ -21,6 +21,10 @@ def create_heading_component_h(data: GTData) -> StringBuilder:
     if not has_title and not has_subtitle:
         return result
 
+    # Raise an error if there is a subtitle but no title
```

Comment:
Added an additional check here to raise in the case that subtitle text is available in the absence of title text. While this may be also caught within the `.tab_header()` method, I think it's good to have a second check here in case there is manual addition/removal of title/subtitle (manipulations done outside the prescribed API).